src/gc-arena/src/allocator_api.rs

```diff
@@ -104,12 +104,7 @@ unsafe impl Collect for Global {
     }
 }
 
-unsafe impl<T: Collect, A: Collect + Allocator> Collect for boxed::Box<T, A> {
-    #[inline]
-    fn needs_trace() -> bool {
-        T::needs_trace() || A::needs_trace()
-    }
-
+unsafe impl<T: Collect + ?Sized, A: Collect + Allocator> Collect for boxed::Box<T, A> {
```

Comment:
It's a little unfortunate that this looses the `needs_trace` optimization, but fixing this would require either a non-object-safe `Collect` trait, or `Collect::needs_trace(&self)`, so oh well...

Reply:
Yeah.. it is unfortunate, but it's the same for the normal `Box` impl too.